### PR TITLE
Added A* algorithm

### LINKED ISSUE #371: [Feature Request] Add A* shortest path algorithm
## 🚀 Feature Request
Add A* shortest path algorithm

### Describe the solution you'd like
I want to add the code for A* shortest path algorithm

---

**Would you like to work on this feature?**  
- [X] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
### Description
Added a start algorithm

Fixes #371 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/graphs/shortest-path-algorithms/A-star-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    # Implement a heuristic function (e.g., Manhattan distance, Euclidean distance)
+    pass
+
+def reconstruct_path(came_from, current):
+    total_path = [current]
+    while current in came_from:
+        current = came_from[current]
+        total_path.append(current)
+    return total_path[::-1]
+
```
Comment (ajay-dhangar) [inline docs/graphs/shortest-path-algorithms/A-star-algorithm.md]: close your code example with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/graphs/shortest-path-algorithms/A-star-algorithm.md (hunk 1 above)
Comment (mehul-m-prajapati) [reply docs/graphs/shortest-path-algorithms/A-star-algorithm.md]: @ajay-dhangar : okay done.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**